src/components/TextField/TextField.tsx

```diff
@@ -18,14 +19,14 @@ export const TextField: React.FC<Props> = ({
   value,
   label = name,
   required = false,
-  onChange = () => {},
+  pattern,
+  onChangeField = () => {},
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
-  const [touched, setToched] = useState(false);
+  const [touched, setTouched] = useState(false);
   const hasError = touched && required && !value;
+  const hasValidURL = pattern ? !value.match(pattern) : false;
```

Comment:
```suggestion
  const hasInvalidURL = pattern ? !value.match(pattern) : false;
```
Судячи зі змісту, така назва буде доречніша